### DISCUSSION TITLE
Loco Positioning with Static objects in room

### DISCUSSION THREAD

**Discussion by afwilkin** [bitcraze/discussions#1062] [Q&A] [answered]
Hello, I was curious about precision of loco in a room with metal static objects. For example,

![image](https://github.com/bitcraze/discussions/assets/16963678/879c4ca2-3264-45e4-80bc-d63b70dfe901)

My concern is that signals might refract off of the static objects, and drastically lower the objects confidence about its position in space. 
Is this a valid concern, or is the margin of error be minimal

**Comment by afwilkin** [comment]
I also just came across this video: https://www.youtube.com/watch?v=TdD82JQrAyk

**Reply by afwilkin** [reply]
so there are some metal pillars in there.

**Reply by krichardsson** [reply, marked as answer]
It is very hard to predict. In my experience it usually works pretty OK also when there are some objects (like in the video) but I guess it depends on the size. The small pillars in the video did not cause any problems, in your case I'm for instance a bit concerned for the top right corner. How high are the objects? If it is full height I doubt it will work behind it.

**Reply by afwilkin** [reply]
They vary in height from waist height to ceiling height. Is it more of a line of sight issue over a refraction issue?

**Reply by krichardsson** [reply]
I think line of sight, but there is a lot of guess work in this answer :-)

**Reply by amitx64** [reply]
@afwilkin Hi, I am also trying to set up as shown in the video. It seems like you've already done one iteration of that. If so, could you please share the steps that I can follow to achieve the same?

**Reply by afwilkin** [reply]
@amitx64 I have not done an iteration so far. Just planning things out. Wanted to better understand if there were any concerns for refraction. I believe it should not be a large issue as long as we have line of sight. At least, that is what I believe as of now. I am sure it will be a bit of trial and error with the environment and calibration as well

**Reply by knmcguire** [reply]
It might be interesting if this work would improve the localization in that sense: https://www.bitcraze.io/2021/06/robust-kalman-filter-for-ultra-wideband-localization/

It is actually intergrated into our firmware. If you take off normally and switch these kalman.robustTdea or robustTwr, perhaps it will improve it even further: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#kalman
